src/plugins/data/public/actions/filters/create_filters_from_value_click.ts

```diff
@@ -148,3 +156,40 @@ export const createFiltersFromValueClickAction = async ({
     compareFilters(a, b, COMPARE_ALL_OPTIONS)
   );
 };
+
+/** @public */
+export const appendFilterToESQLQueryFromValueClickAction = ({
+  data,
+  query,
+}: ValueClickDataContext) => {
+  let queryWithFilter: string | undefined;
+
+  Promise.all(
+    data
+      .filter((point) => point)
+      .map(async (val) => {
+        const { table, column: columnIndex, row: rowIndex } = val;
+        if (table && table.columns && table.columns[columnIndex]) {
+          const column = table.columns[columnIndex];
+          const value: unknown = rowIndex > -1 ? table.rows[rowIndex][column.id] : null;
+
+          // Do not append in case of time series, for now. We need to find a way to compute the interval
+          // to create the time range filter correctly. The users can brush to update the time filter instead.
+          const isDate = column.meta?.type === 'date';
+
+          if (value == null || isDate || !query) return;
+
+          const queryString = query.esql;
+          queryWithFilter = appendWhereClauseToESQLQuery(
+            queryString,
+            column.name,
+            value,
+            '+',
+            column.meta?.type
+          );
+        }
+      })
+  );
```

Comment:
Two things here:
* there's no need of async code at all
* from my understanding only one value is interesting here ( last non-date metric) for the filter

The code currently works as it iterates over all the points computing every time a query, then overwriting the previous one. I think that can be made leaner.

Given the assumptions above, I propose to:
* move the `isDate` check when filtering the `data.points`
* if any metric point is available, pick the last one and build the query with that.

In code I'd suggest something like:

```suggestion
  // no query, no filter
  if(!query){
    return;
  }
  // filter only valid metric points
  const metricPoints = data.filter(
    (point) => point && point.table?.columns?.[point.column].meta?.type !== 'date'
  );

  // if no point is left, exit: no filter added
  if (metricPoints.length === 0) {
    return;
  }
  // otherwise extract the value from the table and build the where clause
  const { table, column: columnIndex, row: rowIndex } = metricPoints[metricPoints.length - 1];
  if (table && table.columns && table.columns[columnIndex]) {
    const column = table.columns[columnIndex];
    const value: unknown = rowIndex > -1 ? table.rows[rowIndex][column.id] : null;

    if (value == null){
      return;
    }
    // return the new query
    return appendWhereClauseToESQLQuery(
      query.esql,
      column.name,
      value,
      '+',
      column.meta?.type
    );
  }
```

Reply:
Makes sense, done https://github.com/elastic/kibana/pull/184420/commits/577c9ae3f36bcbf5f42047d97dae1d92e8c4b08f